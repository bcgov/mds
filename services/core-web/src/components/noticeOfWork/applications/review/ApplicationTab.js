import React, { Component } from "react";
import { Prompt, withRouter } from "react-router-dom";
import { Button, Dropdown, Menu, Popconfirm, Alert, Divider } from "antd";
import { DownOutlined } from "@ant-design/icons";
import PropTypes from "prop-types";
import { getFormValues, reset, getFormSyncErrors, submit, hasSubmitFailed } from "redux-form";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { get, isNull, isUndefined } from "lodash";
import {
  fetchImportedNoticeOfWorkApplication,
  updateNoticeOfWorkApplication,
} from "@common/actionCreators/noticeOfWorkActionCreator";
import { getDropdownInspectors } from "@common/selectors/partiesSelectors";
import {
  getNoticeOfWork,
  getOriginalNoticeOfWork,
  getImportNowSubmissionDocumentsJob,
  getNOWReclamationSummary,
} from "@common/selectors/noticeOfWorkSelectors";
import {
  getGeneratableNoticeOfWorkApplicationDocumentTypeOptions,
  getNoticeOfWorkApplicationStatusOptionsHash,
} from "@common/selectors/staticContentSelectors";
import { flattenObject } from "@common/utils/helpers";
import { downloadNowDocument } from "@common/utils/actionlessNetworkCalls";
import * as Strings from "@common/constants/strings";
import * as Permission from "@/constants/permissions";
import { exportNoticeOfWorkApplicationDocument } from "@/actionCreators/documentActionCreator";
import CustomPropTypes from "@/customPropTypes";
import ReviewNOWApplication from "@/components/noticeOfWork/applications/review/ReviewNOWApplication";
import NOWSideMenu from "@/components/noticeOfWork/applications/NOWSideMenu";
import * as FORM from "@/constants/forms";
import NOWActionWrapper from "@/components/noticeOfWork/NOWActionWrapper";
import NOWTabHeader from "@/components/noticeOfWork/applications/NOWTabHeader";
import AssignInspectors from "@/components/noticeOfWork/applications/verification/AssignInspectors";
import ScrollContentWrapper from "@/components/noticeOfWork/applications/ScrollContentWrapper";
import ReviewAdminAmendmentApplication from "@/components/noticeOfWork/applications/review/ReviewAdminAmendmentApplication";
import { EDIT_OUTLINE } from "@/constants/assets";

/**
 * @constant ApplicationTab renders All content under the Application Tab
 */

const propTypes = {
  updateNoticeOfWorkApplication: PropTypes.func.isRequired,
  fetchImportedNoticeOfWorkApplication: PropTypes.func.isRequired,
  exportNoticeOfWorkApplicationDocument: PropTypes.func.isRequired,
  reset: PropTypes.func.isRequired,
  submit: PropTypes.func.isRequired,
  noticeOfWork: CustomPropTypes.importedNOWApplication.isRequired,
  originalNoticeOfWork: CustomPropTypes.importedNOWApplication.isRequired,
  importNowSubmissionDocumentsJob: PropTypes.objectOf(PropTypes.any).isRequired,
  formValues: CustomPropTypes.importedNOWApplication.isRequired,
  formErrors: PropTypes.objectOf(
    PropTypes.oneOfType([PropTypes.objectOf(PropTypes.string), PropTypes.string])
  ).isRequired,
  fixedTop: PropTypes.bool.isRequired,
  submitFailed: PropTypes.bool.isRequired,
  inspectors: CustomPropTypes.groupOptions.isRequired,
  reclamationSummary: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.strings)).isRequired,
  generatableApplicationDocuments: PropTypes.objectOf(PropTypes.objectOf(PropTypes.any)).isRequired,
  location: PropTypes.shape({
    pathname: PropTypes.string,
    state: PropTypes.shape({
      applicationPageFromRoute: CustomPropTypes.applicationPageFromRoute,
    }),
  }).isRequired,
  isNoticeOfWorkTypeDisabled: PropTypes.bool,
  showActionsAndProgress: PropTypes.bool,
};
const defaultProps = {
  isNoticeOfWorkTypeDisabled: true,
  showActionsAndProgress: true,
};

export class ApplicationTab extends Component {
  state = {
    isInspectorsLoaded: true,
    isViewMode: true,
    menuVisible: false,
    submitting: false,
    submitted: false,
    exportingNow: false,
  };

  count = 1;

  toggleEditMode = () => {
    this.setState((prevState) => ({
      isViewMode: !prevState.isViewMode,
      menuVisible: false,
    }));
  };

  handleMenuClick = () => {
    this.setState({ menuVisible: false });
  };

  exportNowDocument = (documentType) => {
    const documentTypeCode = documentType.now_application_document_type_code;

    const payload = {
      now_application_guid: this.props.noticeOfWork.now_application_guid,
    };
    return this.props.exportNoticeOfWorkApplicationDocument(
      documentTypeCode,
      payload,
      `Successfully exported ${documentType.description} for this Notice of Work`
    );
  };

  // eslint-disable-next-line consistent-return
  handleSaveNOWEdit = (endEditSession) => {
    this.setState({ submitted: true });
    const errors = Object.keys(flattenObject(this.props.formErrors));
    if (errors.length > 0) {
      this.focusErrorInput();
    } else {
      this.setState({ submitting: true });
      return this.props
        .updateNoticeOfWorkApplication(
          this.props.formValues,
          this.props.noticeOfWork.now_application_guid
        )
        .then(() => {
          this.props
            .fetchImportedNoticeOfWorkApplication(this.props.noticeOfWork.now_application_guid)
            .then(() => {
              this.setState(() => ({
                isViewMode: endEditSession,
                submitted: false,
              }));
              if (!endEditSession) {
                // if save & continue - update NoW/form state to reflect changes committed to db
                this.forceUpdate();
              }
            });
        })
        .finally(() => {
          this.setState(() => ({
            submitting: false,
          }));
        });
    }
  };

  focusErrorInput = (skip = false) => {
    this.props.submit(FORM.EDIT_NOTICE_OF_WORK);
    const errors = Object.keys(flattenObject(this.props.formErrors));
    if (skip) {
      if (this.count < errors.length) {
        this.count += 1;
      } else if (this.count === errors.length) {
        this.count = 1;
      }
    }
    const errorElement = document.querySelector(`[name="${errors[this.count - 1]}"]`);
    if (errorElement && errorElement.focus) {
      errorElement.focus();
    }
  };

  handleExportNowDocument = (documentTypeCode) => {
    const documentType = this.props.generatableApplicationDocuments[documentTypeCode];
    this.setState({ exportingNow: true });
    return this.exportNowDocument(documentType, this.props.noticeOfWork)
      .then(() => {
        this.props.fetchImportedNoticeOfWorkApplication(
          this.props.noticeOfWork.now_application_guid
        );
      })
      .finally(() => this.setState({ exportingNow: false }));
  };

  handleVisibleChange = (flag) => {
    this.setState({ menuVisible: flag });
  };

  showApplicationForm = () => {
    const document = this.props.noticeOfWork.submission_documents.filter(
      (x) => x.filename === "ApplicationForm.pdf"
    )[0];
    downloadNowDocument(
      document.id,
      this.props.noticeOfWork.now_application_guid,
      document.filename
    );
  };

  menu = () => {
    const isImported = this.props.noticeOfWork.imported_to_core;
    return (
      <Menu>
        <>
          {isImported &&
            this.props.noticeOfWork.submission_documents.filter(
              (x) => x.filename === "ApplicationForm.pdf"
            ).length > 0 && (
              <Menu.Item
                key="open-original-application-form"
                className="custom-menu-item"
                onClick={this.showApplicationForm}
              >
                Original Application
              </Menu.Item>
            )}
        </>
        <>
          {Object.values(this.props.generatableApplicationDocuments)
            .filter(
              ({ now_application_document_type_code }) =>
                now_application_document_type_code === "NTR"
            )
            .map((document) => {
              return (
                <Menu.Item
                  key={document.now_application_document_type_code}
                  className="custom-menu-item"
                  onClick={() =>
                    this.handleExportNowDocument(document.now_application_document_type_code)
                  }
                  disabled={this.state.exportingNow}
                >
                  Edited Application
                </Menu.Item>
              );
            })}
        </>
      </Menu>
    );
  };

  handleCancelNOWEdit = () => {
    if (this.props.formValues.contacts.length > 0) {
      // eslint-disable-next-line array-callback-return
      this.props.formValues.contacts.map((contact) => delete contact.state_modified);
    }

    this.props.reset(FORM.EDIT_NOTICE_OF_WORK);
    this.setState((prevState) => ({
      isViewMode: !prevState.isViewMode,
    }));
  };

  renderEditModeNav = () => {
    const errorsLength = Object.keys(flattenObject(this.props.formErrors)).length;
    const showErrors = errorsLength > 0 && this.state.submitted && this.props.submitFailed;
    const isNoWApplication = this.props.noticeOfWork.application_type_code === "NOW";
    return (
      <NOWTabHeader
        showProgressButton={this.props.noticeOfWork.lead_inspector_party_guid && isNoWApplication}
        tab="REV"
        showActionsAndProgress={this.props.showActionsAndProgress}
        tabActions={
          this.props.noticeOfWork.lead_inspector_party_guid && (
            <>
              <NOWActionWrapper permission={Permission.EDIT_PERMITS} tab="REV">
                <Button type="secondary" onClick={this.toggleEditMode}>
                  <img alt="EDIT_OUTLINE" className="padding-sm--right" src={EDIT_OUTLINE} />
                  Edit
                </Button>
              </NOWActionWrapper>
              {isNoWApplication && (
                <Dropdown
                  overlay={this.menu(true)}
                  placement="bottomLeft"
                  onVisibleChange={this.handleVisibleChange}
                  visible={this.state.menuVisible}
                >
                  <Button type="secondary" className="full-mobile">
                    Download
                    <DownOutlined />
                  </Button>
                </Dropdown>
              )}
            </>
          )
        }
        tabEditActions={
          <div className="center">
            <Popconfirm
              placement="bottomRight"
              title="You have unsaved changes. Are you sure you want to cancel?"
              onConfirm={this.handleCancelNOWEdit}
              okText="Yes"
              cancelText="No"
              disabled={this.state.submitting}
            >
              <Button type="secondary" className="full-mobile" disabled={this.state.submitting}>
                Cancel
              </Button>
            </Popconfirm>
            {showErrors && (
              <Button
                type="danger"
                className="full-mobile"
                onClick={() => this.focusErrorInput(true)}
              >
                Next Issue
              </Button>
            )}
            <Button
              type="tertiary"
              className="full-mobile"
              onClick={() => this.handleSaveNOWEdit(false)}
              loading={this.state.submitting}
            >
              Save & Continue
            </Button>
            <Button
              type="primary"
              className="full-mobile"
              onClick={() => this.handleSaveNOWEdit(true)}
              loading={this.state.submitting}
            >
              Save
            </Button>
            {showErrors && (
              <Alert
                message={`You have ${errorsLength} ${
                  errorsLength === 1 ? "issue" : "issues"
                } that must be fixed before proceeding.`}
                type="error"
                showIcon
              />
            )}
          </div>
        }
        tabName="Application"
        fixedTop={this.props.fixedTop}
        noticeOfWork={this.props.noticeOfWork}
        isEditMode={!this.state.isViewMode}
      />
    );
  };

  renderOriginalValues = (path, currentPath = null) => {
    const prevValue = get(this.props.originalNoticeOfWork, path);
    const currentValue = get(this.props.noticeOfWork, currentPath || path);
    // cases for isEdited:
    // activities can be added, prevValue === undefined, currentValue === null, thus prevValue !== currentValue - but field has not been edited.
    // prevValue !== undefined || prevValue !==  null, but currentValue has been changed to null, thus is has been edited
    // prevValue !== currentValue, due to other value changes that are not null or undefined
    const isNewValue = isUndefined(prevValue) && !isNull(currentValue);
    const isPrevValue = !isUndefined(prevValue) && !isNull(prevValue);
    const hasBeenEdited = isNewValue || isPrevValue;
    const edited = hasBeenEdited && prevValue !== currentValue;
    const getValue = () => {
      if (prevValue === true) {
        return "Yes";
      }
      if (prevValue === false) {
        return "No";
      }
      if (isUndefined(prevValue) || isNull(prevValue)) {
        return Strings.EMPTY_FIELD;
      }
      return prevValue;
    };

    const toolTipValue = { value: getValue(), edited };
    return toolTipValue;
  };

  handleUpdateInspectors = (values) => {
    this.setState({ isInspectorsLoaded: false });
    return this.props
      .updateNoticeOfWorkApplication(
        values,
        this.props.noticeOfWork.now_application_guid,
        "Successfully updated the assigned inspectors"
      )
      .then(() => {
        this.props
          .fetchImportedNoticeOfWorkApplication(this.props.noticeOfWork.now_application_guid)
          .then(() => this.setState({ isInspectorsLoaded: true }));
      });
  };

  handleResetNOWForm = () => {
    this.props.reset(FORM.EDIT_NOTICE_OF_WORK);
  };

  render() {
    const isImported = this.props.noticeOfWork.imported_to_core;
    const isNoWApplication = this.props.noticeOfWork.application_type_code === "NOW";

    return (
      <React.Fragment>
        <Prompt
          style={{ backgroundColor: "red !important" }}
          when={!this.state.isViewMode}
          message={(location, action) => {
            const onTechnicalReview =
              location.pathname.includes("application") &&
              this.props.location.pathname.includes("application");
            // handle user navigating away from technical review/draft permit while in editMode
            if (action === "REPLACE" && !onTechnicalReview) {
              this.toggleEditMode();
              this.handleResetNOWForm();
            }
            // if the pathname changes while still on the technicalReview tab (via side navigation), don't prompt user
            return this.props.location.pathname === location.pathname && onTechnicalReview
              ? true
              : "You have unsaved changes. Are you sure you want to leave without saving?";
          }}
        />
        {this.renderEditModeNav()}
        <div className={this.props.fixedTop ? "side-menu--fixed" : "side-menu"}>
          <NOWSideMenu tabSection="application" />
        </div>
        <div
          className={
            this.props.fixedTop ? "side-menu--content with-fixed-top" : "side-menu--content"
          }
        >
          {isImported && !this.props.noticeOfWork.lead_inspector_party_guid && (
            <>
              <ScrollContentWrapper id="inspectors" title="Assign Inspectors" isActive>
                <AssignInspectors
                  inspectors={this.props.inspectors}
                  noticeOfWork={this.props.noticeOfWork}
                  handleUpdateInspectors={this.handleUpdateInspectors}
                  title="Assign Inspectors"
                  isEditMode
                  isLoaded={this.state.isInspectorsLoaded}
                />
              </ScrollContentWrapper>
              <Divider />
            </>
          )}
          {isNoWApplication ? (
            <ReviewNOWApplication
              reclamationSummary={this.props.reclamationSummary}
              isViewMode={this.state.isViewMode}
              noticeOfWorkType={this.props.noticeOfWork.notice_of_work_type_code}
              initialValues={this.props.noticeOfWork}
              noticeOfWork={this.props.noticeOfWork}
              importNowSubmissionDocumentsJob={this.props.importNowSubmissionDocumentsJob}
              renderOriginalValues={this.renderOriginalValues}
              isPreLaunch={this.props.originalNoticeOfWork.is_pre_launch}
              isNoticeOfWorkTypeDisabled={this.props.isNoticeOfWorkTypeDisabled}
            />
          ) : (
            <ReviewAdminAmendmentApplication
              reclamationSummary={this.props.reclamationSummary}
              isViewMode={this.state.isViewMode}
              noticeOfWorkType={this.props.noticeOfWork.notice_of_work_type_code}
              initialValues={this.props.noticeOfWork}
              noticeOfWork={this.props.noticeOfWork}
              renderOriginalValues={this.renderOriginalValues}
              isPreLaunch={this.props.originalNoticeOfWork.is_pre_launch}
              isNoticeOfWorkTypeDisabled={this.props.isNoticeOfWorkTypeDisabled}
            />
          )}
        </div>
      </React.Fragment>
    );
  }
}

ApplicationTab.propTypes = propTypes;

const mapStateToProps = (state) => ({
  noticeOfWork: getNoticeOfWork(state),
  originalNoticeOfWork: getOriginalNoticeOfWork(state),
  importNowSubmissionDocumentsJob: getImportNowSubmissionDocumentsJob(state),
  formValues: getFormValues(FORM.EDIT_NOTICE_OF_WORK)(state),
  formErrors: getFormSyncErrors(FORM.EDIT_NOTICE_OF_WORK)(state),
  submitFailed: hasSubmitFailed(FORM.EDIT_NOTICE_OF_WORK)(state),
  inspectors: getDropdownInspectors(state),
  reclamationSummary: getNOWReclamationSummary(state),
  generatableApplicationDocuments: getGeneratableNoticeOfWorkApplicationDocumentTypeOptions(state),
  noticeOfWorkApplicationStatusOptionsHash: getNoticeOfWorkApplicationStatusOptionsHash(state),
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      updateNoticeOfWorkApplication,
      fetchImportedNoticeOfWorkApplication,
      exportNoticeOfWorkApplicationDocument,
      reset,
      submit,
    },
    dispatch
  );

ApplicationTab.propTypes = propTypes;
ApplicationTab.defaultProps = defaultProps;

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(ApplicationTab));
