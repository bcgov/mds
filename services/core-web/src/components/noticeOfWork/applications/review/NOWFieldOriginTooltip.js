import React from "react";
import { Tooltip } from "antd";
import { INFO_CIRCLE } from "@/constants/assets";

export const NOWFieldOriginTooltip = () => (
  <Tooltip
    title="This field is not being sent by NROS or vFCBC. Open the original PDF to to see the data."
    placement="right"
    mouseEnterDelay={0.3}
  >
    <img src={INFO_CIRCLE} alt="Info" className="info-tooltip" />
  </Tooltip>
);

export default NOWFieldOriginTooltip;
