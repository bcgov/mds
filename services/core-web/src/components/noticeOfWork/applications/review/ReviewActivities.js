import React from "react";
import { PropTypes } from "prop-types";
import { bindActionCreators } from "redux";
import { FormSection, arrayInsert, arrayRemove, arrayPush } from "redux-form";
import { connect } from "react-redux";
import * as FORM from "@/constants/forms";
import ScrollContentWrapper from "@/components/common/wrappers/ScrollContentWrapper";
import AccessRoads from "@/components/noticeOfWork/applications/review/activities/AccessRoads";
import Blasting from "@/components/noticeOfWork/applications/review/activities/Blasting";
import SurfaceDrilling from "@/components/noticeOfWork/applications/review/activities/SurfaceDrilling";
import Camps from "@/components/noticeOfWork/applications/review/activities/Camps";
import CutLines from "@/components/noticeOfWork/applications/review/activities/CutLines";
import MechanicalTrenching from "@/components/noticeOfWork/applications/review/activities/MechanicalTrenching";
import SettlingPonds from "@/components/noticeOfWork/applications/review/activities/SettlingPonds";
import SandGravelQuarry from "@/components/noticeOfWork/applications/review/activities/SandGravelQuarry";
import SurfaceBulkSamples from "@/components/noticeOfWork/applications/review/activities/SurfaceBulkSamples";
import WaterSupply from "@/components/noticeOfWork/applications/review/activities/WaterSupply";
import UndergroundExploration from "@/components/noticeOfWork/applications/review/activities/UndergroundExploration";
import Placer from "@/components/noticeOfWork/applications/review/activities/Placer";

/**
 * @constant ReviewActivities renders edit/view for the NoW Application review step
 */

const propTypes = {
  // isViewMode is being passed into field Component, thus ReviewActivities.js assumes it isn't being used
  // eslint-disable-next-line
  isViewMode: PropTypes.bool.isRequired,
  arrayInsert: PropTypes.func.isRequired,
  arrayRemove: PropTypes.func.isRequired,
  arrayPush: PropTypes.func.isRequired,
  // noticeOfWork: CustomPropTypes.importedNOWApplication.isRequired,
};

export const ReviewActivities = (props) => {
  const removeRecord = (activity, rowIndex) => {
    props.arrayRemove(FORM.EDIT_NOTICE_OF_WORK, activity, rowIndex);
  };

  const editRecord = (newActivity, activity, rowIndex) => {
    props.arrayRemove(FORM.EDIT_NOTICE_OF_WORK, activity, rowIndex);
    props.arrayInsert(FORM.EDIT_NOTICE_OF_WORK, activity, rowIndex, newActivity);
  };

  const addRecord = (activity, newActivity) => {
    props.arrayPush(FORM.EDIT_NOTICE_OF_WORK, activity, newActivity);
  };

  return (
    <div>
      <ScrollContentWrapper
        id="access-roads"
        title="Access Roads, Trails, Helipads, Air Strips, Boat Ramps"
      >
        <FormSection name="exploration_access">
          <AccessRoads
            isViewMode={props.isViewMode}
            removeRecord={removeRecord}
            editRecord={editRecord}
            addRecord={addRecord}
          />
        </FormSection>
      </ScrollContentWrapper>
      <ScrollContentWrapper id="blasting" title="Blasting">
        <FormSection name="blasting_operation">
          <Blasting isViewMode={props.isViewMode} />
        </FormSection>
      </ScrollContentWrapper>
      <ScrollContentWrapper
        id="camps"
        title="Camps, Buildings, Staging Areas, Fuel/Lubricant Storage"
      >
        <FormSection name="camps">
          <Camps
            isViewMode={props.isViewMode}
            removeRecord={removeRecord}
            editRecord={editRecord}
            addRecord={addRecord}
          />
        </FormSection>
      </ScrollContentWrapper>
      <ScrollContentWrapper
        id="cut-lines-polarization-survey"
        title="Cut Lines and Induced Polarization Survey"
      >
        <FormSection name="cut_lines_polarization_survey">
          <CutLines
            isViewMode={props.isViewMode}
            removeRecord={removeRecord}
            editRecord={editRecord}
            addRecord={addRecord}
          />
        </FormSection>
      </ScrollContentWrapper>
      <ScrollContentWrapper id="surface-drilling" title="Exploration Surface Drilling">
        <FormSection name="exploration_surface_drilling">
          <SurfaceDrilling
            isViewMode={props.isViewMode}
            removeRecord={removeRecord}
            editRecord={editRecord}
            addRecord={addRecord}
          />
        </FormSection>
      </ScrollContentWrapper>
      <ScrollContentWrapper id="mechanical-trenching" title="Mechanical Trenching / Test Pits">
        <FormSection name="mechanical_trenching">
          <MechanicalTrenching
            isViewMode={props.isViewMode}
            removeRecord={removeRecord}
            editRecord={editRecord}
            addRecord={addRecord}
          />
        </FormSection>
      </ScrollContentWrapper>
      <ScrollContentWrapper id="settling-ponds" title="Settling Ponds">
        <FormSection name="settling_pond">
          <SettlingPonds
            isViewMode={props.isViewMode}
            removeRecord={removeRecord}
            editRecord={editRecord}
            addRecord={addRecord}
          />
        </FormSection>
      </ScrollContentWrapper>
      <ScrollContentWrapper id="surface-bulk-samples" title="Surface Bulk Sample">
        <FormSection name="surface_bulk_sample">
          <SurfaceBulkSamples
            isViewMode={props.isViewMode}
            removeRecord={removeRecord}
            editRecord={editRecord}
            addRecord={addRecord}
          />
        </FormSection>
      </ScrollContentWrapper>
      <ScrollContentWrapper id="underground-exploration" title="Underground Exploration">
        <FormSection name="underground_exploration">
          <UndergroundExploration
            isViewMode={props.isViewMode}
            removeRecord={removeRecord}
            editRecord={editRecord}
            addRecord={addRecord}
          />
        </FormSection>
      </ScrollContentWrapper>
      <ScrollContentWrapper
        id="sand-gravel-quarry-operations"
        title="Sand and Gravel / Quary Operations"
      >
        <FormSection name="sand_and_gravel">
          <SandGravelQuarry
            isViewMode={props.isViewMode}
            removeRecord={removeRecord}
            editRecord={editRecord}
            addRecord={addRecord}
          />
        </FormSection>
      </ScrollContentWrapper>
      <ScrollContentWrapper id="placer-operations" title="Placer Operations">
        <FormSection name="placer_operation">
          <Placer
            isViewMode={props.isViewMode}
            removeRecord={removeRecord}
            editRecord={editRecord}
            addRecord={addRecord}
          />
        </FormSection>
      </ScrollContentWrapper>
      <ScrollContentWrapper id="water-supply" title="Water Supply">
        <FormSection name="water_supply">
          <WaterSupply
            isViewMode={props.isViewMode}
            removeRecord={removeRecord}
            editRecord={editRecord}
            addRecord={addRecord}
          />
        </FormSection>
      </ScrollContentWrapper>
    </div>
  );
};

ReviewActivities.propTypes = propTypes;

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      arrayInsert,
      arrayRemove,
      arrayPush,
    },
    dispatch
  );

export default connect(
  null,
  mapDispatchToProps
)(ReviewActivities);
