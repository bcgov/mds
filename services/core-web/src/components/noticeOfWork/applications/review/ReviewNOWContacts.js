import React from "react";
import { PropTypes } from "prop-types";
import { Row, Col, Card } from "antd";
import NullScreen from "@/components/common/NullScreen";
import CustomPropTypes from "@/customPropTypes";
import * as Strings from "@/constants/strings";
import Address from "@/components/common/Address";
import { formatTitleString } from "@/utils/helpers";

const propTypes = {
  contacts: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.shape({ party: CustomPropTypes.party })))
    .isRequired,
};

const NOWContact = (contact) => (
  <Col key={contact.now_party_appointment_id} xs={24} sm={24} md={12} lg={12} xl={8} xxl={6}>
    {" "}
    <Card
      title={
        <div className="inline-flex between wrap">
          <div>
            <h3>{contact.mine_party_appt_type_code_description}</h3>
          </div>
        </div>
      }
      bordered={false}
    >
      <div>
        <h4>{formatTitleString(contact.party.name)}</h4>
        <br />
        <h6>Email Address</h6>
        {contact.party.email ? (
          <a href={`mailto:${contact.party.email}`}>{contact.party.email}</a>
        ) : (
          <span>{Strings.EMPTY_FIELD}</span>
        )}
        <br />
        <br />
        <h6>Phone Number</h6>
        {contact.party.phone_no} {contact.party.phone_ext ? `x${contact.party.phone_ext}` : ""}
        <br />
        <br />
        <h6>Mailing Address</h6>
        <Address address={contact.party.address.length > 0 && contact.party.address[0]} />
      </div>
    </Card>
  </Col>
);

export const ReviewNOWContacts = (props) => (
  <div>
    {props.contacts && props.contacts.length >= 1 ? (
      <Row gutter={16}>{props.contacts.map((contact) => NOWContact(contact))}</Row>
    ) : (
      <NullScreen type="now-contacts" />
    )}
  </div>
);

ReviewNOWContacts.propTypes = propTypes;

export default ReviewNOWContacts;
