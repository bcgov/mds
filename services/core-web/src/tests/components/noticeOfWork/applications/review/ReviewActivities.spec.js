import React from "react";
import { shallow } from "enzyme";
import { ReviewActivities } from "@/components/noticeOfWork/applications/review/ReviewActivities";
import * as NOW_MOCK from "@/tests/mocks/noticeOfWorkMocks";

const reducerProps = {};

const setupReducerProps = () => {
  reducerProps.isViewMode = true;
  reducerProps.noticeOfWork = NOW_MOCK.IMPORTED_NOTICE_OF_WORK;
  reducerProps.noticeOfWorkType = "COL";
  reducerProps.renderOriginalValues = jest.fn().mockReturnValue({ value: "N/A", edited: true });
};

beforeEach(() => {
  setupReducerProps();
});

describe("ReviewActivities", () => {
  it("renders properly", () => {
    const component = shallow(<ReviewActivities {...reducerProps} />);
    expect(component).toMatchSnapshot();
  });
});
