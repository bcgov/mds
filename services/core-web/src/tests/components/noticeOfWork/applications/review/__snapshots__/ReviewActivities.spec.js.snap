// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewActivities renders properly 1`] = `
<div>
  <withRouter(ScrollContentWrapper)
    id="access-roads"
    title="Access Roads, Trails, Helipads, Air Strips, Boat Ramps"
  >
    <FormSection
      name="exploration_access"
    >
      <Connect(AccessRoads)
        addRecord={[Function]}
        editRecord={[Function]}
        isViewMode={true}
        removeRecord={[Function]}
      />
    </FormSection>
  </withRouter(ScrollContentWrapper)>
  <withRouter(ScrollContentWrapper)
    id="blasting"
    title="Blasting"
  >
    <FormSection
      name="blasting_operation"
    >
      <Blasting
        isViewMode={true}
      />
    </FormSection>
  </withRouter(ScrollContentWrapper)>
  <withRouter(ScrollContentWrapper)
    id="camps"
    title="Camps, Buildings, Staging Areas, Fuel/Lubricant Storage"
  >
    <FormSection
      name="camps"
    >
      <Connect(Camps)
        addRecord={[Function]}
        editRecord={[Function]}
        isViewMode={true}
        removeRecord={[Function]}
      />
    </FormSection>
  </withRouter(ScrollContentWrapper)>
  <withRouter(ScrollContentWrapper)
    id="cut-lines-polarization-survey"
    title="Cut Lines and Induced Polarization Survey"
  >
    <FormSection
      name="cut_lines_polarization_survey"
    >
      <Connect(CutLines)
        addRecord={[Function]}
        editRecord={[Function]}
        isViewMode={true}
        removeRecord={[Function]}
      />
    </FormSection>
  </withRouter(ScrollContentWrapper)>
  <withRouter(ScrollContentWrapper)
    id="surface-drilling"
    title="Exploration Surface Drilling"
  >
    <FormSection
      name="exploration_surface_drilling"
    >
      <Connect(SurfaceDrilling)
        addRecord={[Function]}
        editRecord={[Function]}
        isViewMode={true}
        removeRecord={[Function]}
      />
    </FormSection>
  </withRouter(ScrollContentWrapper)>
  <withRouter(ScrollContentWrapper)
    id="mechanical-trenching"
    title="Mechanical Trenching / Test Pits"
  >
    <FormSection
      name="mechanical_trenching"
    >
      <Connect(MechanicalTrenching)
        addRecord={[Function]}
        editRecord={[Function]}
        isViewMode={true}
        removeRecord={[Function]}
      />
    </FormSection>
  </withRouter(ScrollContentWrapper)>
  <withRouter(ScrollContentWrapper)
    id="settling-ponds"
    title="Settling Ponds"
  >
    <FormSection
      name="settling_pond"
    >
      <Connect(SettlingPonds)
        addRecord={[Function]}
        editRecord={[Function]}
        isViewMode={true}
        removeRecord={[Function]}
      />
    </FormSection>
  </withRouter(ScrollContentWrapper)>
  <withRouter(ScrollContentWrapper)
    id="surface-bulk-samples"
    title="Surface Bulk Sample"
  >
    <FormSection
      name="surface_bulk_sample"
    >
      <Connect(SurfaceBulkSamples)
        addRecord={[Function]}
        editRecord={[Function]}
        isViewMode={true}
        removeRecord={[Function]}
      />
    </FormSection>
  </withRouter(ScrollContentWrapper)>
  <withRouter(ScrollContentWrapper)
    id="underground-exploration"
    title="Underground Exploration"
  >
    <FormSection
      name="underground_exploration"
    >
      <Connect(UndergroundExploration)
        addRecord={[Function]}
        editRecord={[Function]}
        isViewMode={true}
        removeRecord={[Function]}
      />
    </FormSection>
  </withRouter(ScrollContentWrapper)>
  <withRouter(ScrollContentWrapper)
    id="sand-gravel-quarry-operations"
    title="Sand and Gravel / Quary Operations"
  >
    <FormSection
      name="sand_and_gravel"
    >
      <Connect(SandGravelQuarry)
        addRecord={[Function]}
        editRecord={[Function]}
        isViewMode={true}
        removeRecord={[Function]}
      />
    </FormSection>
  </withRouter(ScrollContentWrapper)>
  <withRouter(ScrollContentWrapper)
    id="placer-operations"
    title="Placer Operations"
  >
    <FormSection
      name="placer_operation"
    >
      <Connect(Placer)
        addRecord={[Function]}
        editRecord={[Function]}
        isViewMode={true}
        removeRecord={[Function]}
      />
    </FormSection>
  </withRouter(ScrollContentWrapper)>
  <withRouter(ScrollContentWrapper)
    id="water-supply"
    title="Water Supply"
  >
    <FormSection
      name="water_supply"
    >
      <Connect(WaterSupply)
        addRecord={[Function]}
        editRecord={[Function]}
        isViewMode={true}
        removeRecord={[Function]}
      />
    </FormSection>
  </withRouter(ScrollContentWrapper)>
</div>
`;
